### PR TITLE
Make Event interface.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/trace/SpanData.java [hunk 1]
```diff
@@ -175,6 +175,46 @@ public static SpanData create(
...
+     * @param name the text name of the {@code Event}.
+     * @param attributes the attributes of the {@code Event}.
+     * @return a new {@code Event} with the given name and set of attributes.
+     * @throws NullPointerException if {@code name} or {@code attributes} are {@code null}.
+     * @since 0.1.0
+     */
+    public static io.opentelemetry.trace.Event create(
+        String name, Map<String, AttributeValue> attributes) {
+      return new AutoValue_SpanData_Event(
+          name,
```
Comment (pavolloffay) [inline api/src/main/java/io/opentelemetry/trace/SpanData.java]: check for null?

File: api/src/main/java/io/opentelemetry/trace/SpanData.java [hunk 2]
```diff
@@ -175,6 +175,46 @@ public static SpanData create(
...
+    /**
+     * Returns a new {@code Event} with the given name.
+     *
+     * @param name the text name of the {@code Event}.
+     * @return a new {@code Event} with the given name.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @since 0.1.0
+     */
+    public static io.opentelemetry.trace.Event create(String name) {
+      return new AutoValue_SpanData_Event(name, EMPTY_ATTRIBUTES);
```
Comment (pavolloffay) [inline api/src/main/java/io/opentelemetry/trace/SpanData.java]: check name for null?

Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/SpanData.java]: Autovalue generated classes will do that for us.

File: api/src/main/java/io/opentelemetry/trace/SpanData.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/trace/SpanData.java]: Same, autovalue generated classes do that.

**Pull request merged**